sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -118,15 +122,25 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
   }
 
   /**
-   * Check if education program is active.
+   * Check if education program is active and not expired.
    * @param educationProgramId program id.
    */
-  async isProgramActive(educationProgramId: number): Promise<boolean> {
+  async isProgramActiveAndNotExpired(
+    educationProgramId: number,
+  ): Promise<boolean> {
     return this.repo.exists({
-      where: {
-        id: educationProgramId,
-        isActive: true,
-      },
+      where: [
+        {
+          id: educationProgramId,
+          isActive: true,
+          effectiveEndDate: MoreThan(getISODateOnlyString(new Date())),
```

Comment:
Would this work for Programs where `effectiveEndDate` is `null` or should an `OR` be added here?

Reply:
The check for when the `effectiveEndDate`  is null in lines 138-142. The `OR` in the object query requires an array for the where clause: https://github.com/typeorm/typeorm/blob/master/docs/find-options.md
![image](https://github.com/bcgov/SIMS/assets/78114138/1de6dc67-2610-4b8b-a901-a548c6ef3000)

Reply:
I missed the change in the code and I got the change. Let's wait for the test with the typeorm "Or" to see if it is available in our current version.